File: 2023/Leetcode/algorithms/easy/0012_integer_to_roman/0012_integer_to_roman.py
class Solution:
    def intToRoman(self, num: int) -> str:
        val = [
            1000, 900, 500, 400,
            100, 90, 50, 40,
            10, 9, 5, 4,
            1
        ]
        syms = [
            "M", "CM", "D", "CD",
            "C", "XC", "L", "XL",
            "X", "IX", "V", "IV",
            "I"
        ]
        roman_numeral = ''
        i = 0
        
        while num > 0:
            for _ in range(num // val[i]):
                roman_numeral += syms[i]
                num -= val[i]
            i += 1
        
        return roman_numeral

File: 2023/Leetcode/algorithms/easy/0012_integer_to_roman/0012_integer_to_roman.ts
function intToRoman(num: number): string {
    if (num < 1 || num > 3999) {
        return ""; // Return an empty string for out-of-range numbers
    }
    
    const val: number[] = [
        1000, 900, 500, 400,
        100, 90, 50, 40,
        10, 9, 5, 4,
        1,
    ];
    const syms: string[] = [
        "M", "CM", "D", "CD",
        "C", "XC", "L", "XL",
        "X", "IX", "V", "IV",
        "I",
    ];
    
    let romanNumeral: string = "";
    let i: number = 0;
    
    while (num > 0) {
        while (num >= val[i]) {
            romanNumeral += syms[i];
            num -= val[i];
        }
        i++;
    }
    
    return romanNumeral;
}
